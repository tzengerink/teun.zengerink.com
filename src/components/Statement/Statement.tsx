import classNames from 'classnames'
import Markdown from '../Markdown/Markdown'
import { useState } from 'react'

interface StatementProps {
  title: string
  statement: string
}

const Statement: React.FC<StatementProps> = ({ title, statement }) => {
  const [isOpen, setIsOpen] = useState(false)

  return (
    <div className={classNames('group', 'text-grey/80', 'select-none')}>
      <h2
        className={classNames('fixed', 'left-md', 'bottom-md', 'z-20', 'text-xs', 'uppercase', 'cursor-help')}
        onClick={() => setIsOpen(!isOpen)}
      >
        Statement
      </h2>
      <div
        className={classNames(
          'fixed',
          'left-0',
          'bottom-md',
          'z-10',
          'right-0',
          'mb-md',
          'p-md',
          'bg-transparant-white',
          { invisible: !isOpen },
          { 'opacity-0': !isOpen },
          'ease-linear',
          'duration-100',
          { 'translate-y-[50px]': !isOpen },
          'group-hover:visible',
          'group-hover:translate-y-0',
          'group-hover:opacity-100',
          'md:w-[500px]',
        )}
      >
        <h3 className={classNames('mb-sm', 'text-sm')}>{title}</h3>
        <Markdown>{statement}</Markdown>
      </div>
    </div>
  )
}

export default Statement
