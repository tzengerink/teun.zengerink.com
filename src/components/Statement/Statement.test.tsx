import { render } from '@testing-library/react'
import Statement from './Statement'

describe('Statement', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Statement title="Hello World!" statement="This is a _message_ for the **entire world** to see" />,
    )
    expect(container).toMatchSnapshot()
  })
})
