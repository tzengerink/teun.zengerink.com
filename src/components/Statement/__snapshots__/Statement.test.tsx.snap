// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Statement renders correctly 1`] = `
<div>
  <div
    class="group text-grey/80 select-none"
  >
    <h2
      class="fixed left-md bottom-md z-20 text-xs uppercase cursor-help"
    >
      Statement
    </h2>
    <div
      class="fixed left-0 bottom-md z-10 right-0 mb-md p-md bg-transparant-white invisible opacity-0 ease-linear duration-100 translate-y-[50px] group-hover:visible group-hover:translate-y-0 group-hover:opacity-100 md:w-[500px]"
    >
      <h3
        class="mb-sm text-sm"
      >
        Hello World!
      </h3>
      <div
        class="space-y-sm text-xs text-justify"
      >
        <p>
          This is a 
          <em>
            message
          </em>
           for the 
          <strong>
            entire world
          </strong>
           to see
        </p>
      </div>
    </div>
  </div>
</div>
`;
